lightningd/plugin.c

```diff
@@ -1385,6 +1385,59 @@ static struct command_result *plugin_rpcmethod_dispatch(struct command *cmd,
 	return command_still_pending(cmd);
 }
 
+static const char *plugin_rpcmethod_clnrest_add(struct plugin *plugin, 
+                                                const char *buffer, 
+                                                const jsmntok_t *tok, 
+                                                struct json_command *cmd) 
+{
+	const jsmntok_t *clnresttok = json_get_member(buffer, tok, "clnrest");
+
+	if (!clnresttok) {
+			cmd->clnrest = NULL;
+			return NULL;
+	}
+
+	struct clnrest *clnrest = tal(cmd, struct clnrest);
+
+	// TODO: check each field for validity
+
+	const jsmntok_t *pathtok = json_get_member(buffer, clnresttok, "path");
+	const jsmntok_t *methodtok = json_get_member(buffer, clnresttok, "method");
+	const jsmntok_t *contenttypetok = json_get_member(buffer, clnresttok, "content_type");
+	const jsmntok_t *runetok = json_get_member(buffer, clnresttok, "rune");
+	const jsmntok_t *pluginnametok = json_get_member(buffer, tok, "name");
+
+	if (pathtok) {
+		clnrest->path = json_strdup(clnrest, buffer, pathtok);
+	} else {
+		clnrest->path = json_strdup(clnrest, buffer, pluginnametok);
```

Comment:
nit: i think technically the 'name' here is the name of the plugin command, not the plugin?

might be worthwhile to add a note somewhere that it defaults to the command name.

am i right in thinking this will make it such that clnrest will dynamically add a GET endpoint for every new plugin method that's added?? incredible.

Reply:
note that if that's the case we might have some default /v1/ etc prefix to add to the name?

Reply:
> 'name' here is the name of the plugin command, not the plugin
oo yea you're right

> am i right in thinking this will make it such that clnrest will dynamically add a GET endpoint for every new plugin method that's added?? 

Only if that method includes clnrest data in response to `get_manifest`. A plugin might want to specify the return content type, but nothing else...